gwsurrogate/surrogateIO.py

```diff
@@ -558,6 +558,31 @@ def phase_fit_func(coeffs,xvec):
     
     if closeQ:
       self.file.close()
+
+
+
+    ### check that data has been loaded as 64-bit double (NOTE: not all datasets are checked here)
```

Comment:
Will this now be done for all surrogates that use `H5Surrogate`?
Is it better to add a special option for the EMRI1dSur?
For example, in the spinning version, we will probably just fix this at the h5 file stage.

Reply:
Indeed, all of the surrogates using H5Surrogate will be checked. This was actually intentional, in case X years from now we forget about the issue and someone makes a surrogate with single-precision datasets.  Its also the case that maybe single precision is OK for saving the data, but not ideal for arithmetic operations. So this will do the necessary upcasting as well as alerting the user to this.

So I guess I'm slightly in favor of keeping the more general check in place. But since its only one hdf5 dataset affected at the moment, we could make the if statement specific to EMRI1dSur if you think this is preferable.

Reply:
Ok, I also like keeping it, but just to make sure this is a one-time check that only happens when loading the data, right?

Reply:
exactly. One-time check only when loading.